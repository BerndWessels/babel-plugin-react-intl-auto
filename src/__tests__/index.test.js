// @flow
import path from 'path'
import fs from 'fs'
import { transformFileSync } from 'babel-core'

import plugin from '../'

const testPlugin = (filePath, opts = {}) => {
  const { code } = transformFileSync(filePath, {
    plugins: [[plugin, opts]],
  })
  return code
}

describe('snapshot', () => {
  const rootPath = path.resolve(__dirname, '../__fixtures__')
  const files = fs.readdirSync(rootPath)

  for (const file of files) {
    test(`snapshot -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file))
      expect(result).toMatchSnapshot()
    })

    test(`removePrefix -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        removePrefix: 'src/',
      })
      expect(result).toMatchSnapshot()
    })

    test(`removePrefix:no-slash -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        removePrefix: 'src',
      })
      expect(result).toMatchSnapshot()
    })

    test(`filebase = true -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        filebase: true,
      })
      expect(result).toMatchSnapshot()
    })

    test(`filebase = false -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        filebase: false,
      })
      expect(result).toMatchSnapshot()
    })

    test(`includeExportName = true -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        includeExportName: true,
      })
      expect(result).toMatchSnapshot()
    })

    test(`includeExportName = all -- ${file}`, () => {
      const result = testPlugin(path.join(rootPath, file), {
        includeExportName: 'all',
      })
      expect(result).toMatchSnapshot()
    })
  }
})
