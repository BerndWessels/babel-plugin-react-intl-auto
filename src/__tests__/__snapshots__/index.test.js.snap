// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot filebase = false -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = false -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = false -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = false -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot filebase = false -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot filebase = false -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot filebase = false -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot filebase = true -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.basic.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.basic.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = true -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.import-as.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.import-as.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = true -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.multi-export.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.multi-export.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.multi-export.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.multi-export.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot filebase = true -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.object.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.object.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.object.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot filebase = true -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.string.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.string.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot filebase = true -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': 'src.__fixtures__.string-literal.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': 'src.__fixtures__.string-literal.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot filebase = true -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.with-other-func.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.with-other-func.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot includeExportName = all -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = all -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = all -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.extra.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.extra.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.default.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.default.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = all -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot includeExportName = all -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot includeExportName = all -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot includeExportName = all -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot includeExportName = true -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = true -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = true -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.extra.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.extra.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot includeExportName = true -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot includeExportName = true -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot includeExportName = true -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot includeExportName = true -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot removePrefix -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': '__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': '__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot removePrefix -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot removePrefix -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot removePrefix -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot removePrefix:no-slash -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': '__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': '__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot removePrefix:no-slash -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot removePrefix:no-slash -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;

exports[`snapshot snapshot -- basic.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot snapshot -- import-as.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot snapshot -- multi-export.js 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.extra = undefined;

var _reactIntl = require('react-intl');

const extra = exports.extra = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

exports.default = (0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});"
`;

exports[`snapshot snapshot -- object.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});"
`;

exports[`snapshot snapshot -- string.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});"
`;

exports[`snapshot snapshot -- string-literal.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  'hello': {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  }, // prettier-ignore
  'world': {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  } // prettier-ignore
});"
`;

exports[`snapshot snapshot -- with-other-func.js 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});"
`;
